internal/cli/command/controlplane/debug.go

```diff
@@ -0,0 +1,241 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"emperror.dev/errors"
+	"github.com/banzaicloud/banzai-cli/internal/cli"
+	"github.com/shirou/gopsutil/host"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+const basePath = "pipeline-debug-bundle/"
+
+type debugOptions struct {
+	filename string
+	*cpContext
+}
+
+func (d *debugOptions) filepath() string {
+	if !d.cpContext.flags.Changed("filename") {
+		d.filename = fmt.Sprintf("pipeline-debug-bundle-%s.tgz", time.Now().Format("20060102-1504"))
```

Comment:
I see that, but I still not see why it indicates a custom format is better than a standard one.
Also as a user, eyeballing 8 numbers next to each other - unless the context is sorted accordingly - is not really a convenient way to compare files so I think for example `YYYY-MM-DD` is better than `YYYYMMDD` at helping the end user finding the correct bundle to send.

Reply:
I wanted to make it shorter than common formats because I don't like long file names with the significant difference at the end, because in many cases (for example when you browse it for upload) you will see it trimmed down.

Reply:
I agree, that's where putting the timestamp at the beginning might help.

Reply:
As we discussed earlier, `:` is invalid in file names on Windows, and this file is something that may be submitted from a Windows workstation at the end, so I'd keep it compatible. Without the colon I see no more human-readable timestamp formats that are in the right order if sorted lexicographically. `2020-09-04-12-39` or `2020-09-04-1239` or `2020-09-04T12-39` seem nothing better for me compared to the shorter `20200904-1239`. None of these are standard afaik, and RFC3339 requires the colon if a time is included anyway.